src/common/post.ts

```diff
@@ -534,3 +539,87 @@ export const processApprovedModeratedPost = async (
 
   return null;
 };
+
+export const validateSourcePostModeration = async (
+  ctx: AuthContext,
+  {
+    postId,
+    title,
+    content,
+    sourceId,
+    image,
+    type,
+    sharedPostId,
+    imageUrl,
+    externalLink,
+  }: CreateSourcePostModerationArgs,
+): Promise<CreateSourcePostModeration> => {
+  if (![PostType.Share, PostType.Freeform].includes(type)) {
+    throw new ValidationError('Invalid post type!');
+  }
+
+  const { con, userId } = ctx;
+
+  const sourceMember = await con.getRepository(SourceMember).findOne({
+    where: { sourceId: sourceId, userId: userId },
+    relations: ['source'],
+  });
+
+  const source = await sourceMember?.source;
```

Comment:
Can this be passed from outside, `validateSourcePostModeration` could be easily reused and then we will just be querying the DB all over the place. Let's keep the permissions/source check directly in mutations/queries

Reply:
I've removed some things that should be automatic when we run `ensureSourcePermission`.